src/seed_clusterer.cpp

```diff
@@ -165,19 +163,24 @@ for (pair<id_t, vector<size_t>> n : node_to_seed){
 #endif
   
         // This will hold all top-level snarls that have seeds under them
-        vector<const Snarl*> top_snarls;
-        for (auto& snarl : snarls_to_node) {
-            if (snarl_manager.parent_of(snarl.first) == nullptr) {
-                // This is top level
-                top_snarls.push_back(snarl.first);
-            }
-        }
+        //vector<const Snarl*> top_snarls;
+        //for (auto& snarl : snarls_to_node) {
+        //    if (snarl_manager.parent_of(snarl.first) == nullptr) {
+        //        // This is top level
+        //        top_snarls.push_back(snarl.first);
+        //    }
+        //}
         
         // We track seen-ness by chain, so we don't have to do O(N) work to tag
         // all snarls in chromosome-spanning chains as seen.
         unordered_set<const Chain*> seen_chains;
         vector<hash_set<size_t>> cluster_assignments; 
-        for (const Snarl* root_snarl : top_snarls) {
+        //for (const Snarl* root_snarl : top_snarls) {
+        for (auto& snarl : snarls_to_node) {
```

Comment:
Get top level snarls from SnarlManager instead of iterating all snarls and continuing when it's not a root.